### PR TITLE
Fixed: API Level 30 Migrations

### PR DESCRIPTION
**Pull request by akshaaatt**
- Removed the deprecated Environment.getExternalStorageDirectory() which crashes the platform and replaced it with Environment.getRootDirectory() implementation.

- With the update, removed the permission requirements for WRITE_EXTERNAL_STORAGE since it is no longer needed.

- Removed the kotlin-android-extensions since it has now been deprecated.

- Replaced the deprecated Preference Manager implementation with the Android X implementation.

- Tested and all seems to work fine.

- Available to fix more issues and updating if need be.

### CODE REVIEW

File: app/src/main/java/org/metabrainz/mobile/presentation/UserPreferences.java [hunk 1]
```diff
@@ -19,9 +19,10 @@
     private static final String PREFERENCE_RATINGS_TAGS = "ratings_tags";
     private static final String PREFERENCE_SYSTEM_LANGUAGE = "use_english";
     private static final String PREFERENCE_ONBOARDING = "show_onboarding";
+    private static final String SHARED_PREF_FILE = "shared_pref_file";
 
     private static SharedPreferences getPreferences() {
-        return PreferenceManager.getDefaultSharedPreferences(App.getContext());
+        return App.getContext().getSharedPreferences(SHARED_PREF_FILE, Context.MODE_PRIVATE);
```
Comment (amCap1712) [inline app/src/main/java/org/metabrainz/mobile/presentation/UserPreferences.java:25]: The [PreferenceManagerCompat](https://developer.android.com/reference/androidx/preference/PreferenceFragmentCompat) documentation mentions:
>To retrieve an instance of SharedPreferences that the preference hierarchy in this fragment will use by default, call PreferenceManager.getDefaultSharedPreferences(android.content.Context) with a context in the same package as this fragment.

Hence, replacing the `getDefaultSharedPreferences` would also need changing the `SettingsFragment` to use it.

Another solution is to use the AndroidX Preference Library as this [deprecation note](https://developer.android.com/reference/android/preference/PreferenceManager) mentions.

**Review by amCap1712 — CHANGES_REQUESTED** [review]
Thanks for the contribution. The Preferences change needs some fixing, see the inline comment.

File: app/src/main/java/org/metabrainz/mobile/presentation/UserPreferences.java (hunk 1 above)
Comment (akshaaatt) [reply app/src/main/java/org/metabrainz/mobile/presentation/UserPreferences.java:25]: Thanks a lot for this suggestion! I will work on it straightaway! :)

**Comment by akshaaatt** [comment]
I have made the changes @amCap1712 as you requested. 
Thank you.

**Review by amCap1712 — APPROVED** [review]

**Comment by amCap1712** [comment]
Looks great, thanks!

**Pull request merged**